src/uu/whereis/src/whereis.rs

```diff
@@ -0,0 +1,403 @@
+// This file is a part of the uutils util-linux package.
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use clap::{crate_version, Arg, ArgAction, Command};
+use glob::glob;
+use serde::Serialize;
+use std::{
+    collections::HashMap, collections::HashSet, fs, os::unix::fs::MetadataExt, path::Path,
+    path::PathBuf,
+};
+use uucore::{error::UResult, format_usage, help_about, help_usage};
+
+mod constants;
+use crate::constants::{BINDIRS, MANDIRS, SRCDIRS};
+
+mod options {
+    pub const BIN: &str = "binaries";
+    pub const MAN: &str = "manuals";
+    pub const SRC: &str = "sources";
+    pub const PATH: &str = "lookups";
+
+    pub const SPECIFIED_BIN: &str = "listed binaries";
+    pub const SPECIFIED_MAN: &str = "listed manuals";
+    pub const SPECIFIED_SRC: &str = "listed sources";
+}
+
+const ABOUT: &str = help_about!("whereis.md");
+const USAGE: &str = help_usage!("whereis.md");
+
+// Directories are usually manual pages dirs, binary dirs or source dirs. Hopefully not unknown.
+#[derive(Serialize, Copy, Clone, Debug, Eq, Hash, PartialEq)]
+pub enum DirType {
+    BIN,
+    MAN,
+    SRC,
+    UNK,
```

Comment:
The names of enum variants usually use UpperCamelCase. I would also use `Unknown` instead of `UNK`.